### PR TITLE
Bugfix/fix terraform refresh processing of manually deleted resources

### PR DESCRIPTION
**Pull request by iclonghoang**
**Scenario**:
resources created and tracked by terraform can be manually deleted by provider API

**Expectation**:
terraform apply can detect resources deletion and restore them
**Issue**:
terraform refresh state ignores provider API response with 404 Not Found
hence unable to detect/restore deleted resources

**Fix**:
terraform refresh state to handle 404 Not Found properly

### CODE REVIEW

File: go.mod
Comment (iclonghoang) [inline go.mod:1]: auto updated by build process

File: go.sum
Comment (iclonghoang) [inline go.sum:1]: auto updated by build process

File: openapi/resource_factory.go
Comment (iclonghoang) [inline openapi/resource_factory.go:1]: actual fix to empty resource in state when getting 404 response

File: tests/e2e/gray_box_cdns_test.go
Comment (iclonghoang) [inline tests/e2e/gray_box_cdns_test.go:1]: test handling of 404 response by `terraform plan`

**Review by arjunrajnetapp — APPROVED** [review]

**Pull request merged**